fhirclient/client.py

```diff
@@ -240,3 +243,97 @@ def from_state(self, state):
+    def fetch_next_page(self, bundle):
+        """
+        Fetch the next page of results using the `next` link provided in the bundle.
+
+        Args:
+            bundle (Bundle): The FHIR Bundle containing the `next` link.
+
+        Returns:
+            Bundle: The next page of results as a FHIR Bundle.
+        """
+        next_link = self.get_next_link(bundle)
+        if next_link:
+            sanitized_next_link = self.sanitize_next_link(next_link)
+            return self.execute_pagination_request(sanitized_next_link)
+        return None
+
+    def get_next_link(self, bundle):
+        """
+        Extract the `next` link from the Bundle's links.
+
+        Args:
+            bundle (Bundle): The FHIR Bundle containing pagination links.
+
+        Returns:
+            str: The URL of the next page if available, None otherwise.
+        """
+        for link in bundle["link"]:
+            if link["relation"] == "next":
+                return link["url"]
+        return None
+
+    def sanitize_next_link(self, next_link):
+        """
+        Sanitize the `next` link to ensure it is safe to use.
+
+        Args:
+            next_link (str): The raw `next` link URL.
+
+        Returns:
+            str: The sanitized URL.
+
+        Raises:
+            ValueError: If the URL scheme or domain is invalid.
+        """
+        parsed_url = urllib.parse.urlparse(next_link)
+
+        # Validate scheme and netloc (domain)
+        if parsed_url.scheme not in ["http", "https"]:
+            raise ValueError("Invalid URL scheme in `next` link.")
+        if not parsed_url.netloc:
+            raise ValueError("Invalid URL domain in `next` link.")
+
+        # Additional sanitization if necessary, e.g., removing dangerous query parameters
+        query_params = urllib.parse.parse_qs(parsed_url.query)
+        sanitized_query = {k: v for k, v in query_params.items()}
+
+        # Rebuild the sanitized URL
+        sanitized_url = urllib.parse.urlunparse(
+            (
+                parsed_url.scheme,
+                parsed_url.netloc,
+                parsed_url.path,
+                parsed_url.params,
+                urllib.parse.urlencode(sanitized_query, doseq=True),
+                parsed_url.fragment,
+            )
+        )
+
+        return sanitized_url
+
+    def execute_pagination_request(self, sanitized_url):
+        """
+        Execute the request to retrieve the next page using the sanitized URL.
+
+        Args:
+            sanitized_url (str): The sanitized URL to fetch the next page.
+
+        Returns:
+            Bundle: The next page of results as a FHIR Bundle.
+
+        Raises:
+            HTTPError: If the request fails due to network issues or server errors.
+        """
+        try:
+            # Use requests.get directly to make the HTTP request
+            response = requests.get(sanitized_url)
+            response.raise_for_status()
+            return response.json()
```

Comment:
Shouldn't this instantiate a Bundle object?

Reply:
Thank you! I'll make changes and push edited code shortly.